packages/agw-client/src/actions/sendPrivyTransaction.ts

```diff
@@ -65,75 +34,23 @@ export async function sendPrivyTransaction<
   chain extends ChainEIP712 | undefined = ChainEIP712 | undefined,
   account extends Account | undefined = Account | undefined,
   chainOverride extends ChainEIP712 | undefined = ChainEIP712 | undefined,
+  const request extends SendTransactionRequest<
+    chain,
+    chainOverride
+  > = SendTransactionRequest<chain, chainOverride>,
 >(
   client: Client<Transport, ChainEIP712, Account>,
-  signerClient: WalletClient<Transport, ChainEIP712, Account>,
-  args: SignEip712TransactionParameters<chain, account, chainOverride>,
-  useSignerAddress = false,
+  parameters:
+    | SendEip712TransactionParameters<chain, account, chainOverride, request>
+    | SendTransactionBatchParameters<request>,
 ): Promise<SignEip712TransactionReturnType> {
-  const {
-    account: account_ = client.account,
-    chain = client.chain,
-    ...transaction
-  } = args;
-  // TODO: open up typing to allow for eip712 transactions
-  transaction.type = 'eip712' as any;
-  transaction.value = transaction.value
-    ? BigInt(transaction.value)
-    : (0n as any);
-
-  if (!account_)
-    throw new AccountNotFoundError({
-      docsPath: '/docs/actions/wallet/signTransaction',
-    });
-  const smartAccount = parseAccount(account_);
-  const fromAccount = useSignerAddress ? signerClient.account : smartAccount;
-
-  assertEip712Request({
-    account: smartAccount,
-    chain,
-    ...(transaction as AssertEip712RequestParameters),
-  });
-
-  if (!chain || !ALLOWED_CHAINS.includes(chain.id)) {
-    throw new BaseError('Invalid chain specified');
-  }
-
-  if (!chain?.custom?.getEip712Domain)
-    throw new BaseError('`getEip712Domain` not found on chain.');
-  if (!chain?.serializers?.transaction)
-    throw new BaseError('transaction serializer not found on chain.');
-
-  const chainId = await getAction(client, getChainId, 'getChainId')({});
-  if (chain !== null)
-    assertCurrentChain({
-      currentChainId: chainId,
-      chain: chain,
-    });
-
-  const calls: Call[] = [
-    {
-      target: transaction.to as Address,
-      allowFailure: false,
-      value: BigInt(transaction.value ?? 0),
-      callData: transaction.data ?? '0x',
-    },
-  ];
-
-  const serializedTransaction = convertBigIntToString(transaction);
-
   const result = (await client.request(
     {
       method: 'privy_sendSmartWalletTx',
-      params: [
-        fromAccount,
-        serializedTransaction,
-        convertCallsToPermissionlessCalls(calls),
-      ],
+      params: [convertBigIntToString(parameters)],
```

Comment:
**suggestion:** doesn't need to be this PR but lets yoink [`replaceBigInts` from ponder](https://github.com/ponder-sh/ponder/blob/bc8eefaf3c044f77b0f751315105907891125780/packages/utils/src/replaceBigInts.ts) - this comes with better typing and lets you specify the output with a delegate function